plugins/webp-uploads/hooks.php

```diff
@@ -681,41 +681,58 @@ function webp_uploads_update_featured_image( $html, $post_id, $attachment_id ) {
-	if ( ! $preserve_jpegs_for_jpeg_transforms && ! $preserve_jpegs_for_webp_transforms ) {
+	// We need to add fallback only if jpeg alternatives for the image_format images are enabled for the server.
+	$preserve_jpegs_for_jpeg_transforms       = isset( $transforms['image/jpeg'] ) && in_array( 'image/jpeg', $transforms['image/jpeg'], true ) && in_array( 'image/' . $image_format, $transforms['image/jpeg'], true );
+	$preserve_jpegs_for_image_type_transforms = isset( $transforms[ 'image/' . $image_format ] ) && in_array( 'image/jpeg', $transforms[ 'image/' . $image_format ], true ) && in_array( 'image/' . $image_format, $transforms[ 'image/' . $image_format ], true );
+	if ( ! $preserve_jpegs_for_jpeg_transforms && ! $preserve_jpegs_for_image_type_transforms ) {
 		return;
 	}
 
-	ob_start();
+	$detection_string = '';
+	// The fallback script can only handle a single image format at a time.
+	if ( 'webp' === $image_format ) {
+		$detection_string = 'data:image/webp;base64,UklGRh4AAABXRUJQVlA4TBEAAAAvAQAAAAfQ//73v/+BiOh/AAA=';
+	} elseif ( 'avif' === $image_format ) {
+		$detection_string = 'data:image/avif;base64,AAAAIGZ0eXBhdmlmAAAAAGF2aWZtaWYxbWlhZk1BMUIAAADybWV0YQAAAAAAAAAoaGRscgAAAAAAAAAAcGljdAAAAAAAAAAAAAAAAGxpYmF2aWYAAAAADnBpdG0AAAAAAAEAAAAeaWxvYwAAAABEAAABAAEAAAABAAABGgAAAB0AAAAoaWluZgAAAAAAAQAAABppbmZlAgAAAAABAABhdjAxQ29sb3IAAAAAamlwcnAAAABLaXBjbwAAABRpc3BlAAAAAAAAAAIAAAACAAAAEHBpeGkAAAAAAwgICAAAAAxhdjFDgQ0MAAAAABNjb2xybmNseAACAAIAAYAAAAAXaXBtYQAAAAAAAAABAAEEAQKDBAAAACVtZGF0EgAKCBgANogQEAwgMg8f8D///8WfhwB8+ErK42A=';
+	}
+	if ( '' === $detection_string ) {
+		return;
+	}
 
-	?>
-	( function( d, i, s, p ) {
+	$js_function = <<<JS
+	/**
+	 * Detect if the browser supports the current output format and if not loads the fallback.js script.
+	 */
+	function webpUploadsDetectFallback( d, i, s, p, fallbackSrc ) {
 		s = d.createElement( s );
-		s.src = '<?php echo esc_url_raw( plugins_url( '/fallback.js', __FILE__ ) ); ?>';
+		s.src = fallbackSrc;
 
 		i = d.createElement( i );
-		i.src = p + 'jIAAABXRUJQVlA4ICYAAACyAgCdASoCAAEALmk0mk0iIiIiIgBoSygABc6zbAAA/v56QAAAAA==';
+		i.src = p;
 		i.onload = function() {
 			i.onload = undefined;
-			i.src = p + 'h4AAABXRUJQVlA4TBEAAAAvAQAAAAfQ//73v/+BiOh/AAA=';
 		};
 
 		i.onerror = function() {
 			d.body.appendChild( s );
 		};
-	} )( document, 'img', 'script', 'data:image/webp;base64,UklGR' );
-	<?php
-	$javascript = ob_get_clean();
+	}
+JS;
 
 	wp_print_inline_script_tag(
-		preg_replace( '/\s+/', '', $javascript ),
+		sprintf(
+			'( %s )( document, "img", "script", %s, %s )',
+			preg_replace( '/\s+/', ' ', $js_function ),
+			wp_json_encode( $detection_string ),
+			wp_json_encode( plugins_url( '/fallback.js', __FILE__ ) )
+		),
 		array(
 			'id'            => 'webpUploadsFallbackWebpImages',
 			'data-rest-api' => esc_url_raw( trailingslashit( get_rest_url() ) ),
+			'data-output-format' => webp_uploads_get_image_output_format(),
```

Comment:
My bad. I made PHPCS unhappy.

```suggestion
			'id'                 => 'webpUploadsFallbackWebpImages',
			'data-rest-api'      => esc_url_raw( trailingslashit( get_rest_url() ) ),
			'data-output-format' => webp_uploads_get_image_output_format(),
```